src/app/pages/categories/management/pages-categories-management.component.ts

```diff
@@ -78,6 +87,50 @@ export class PagesCategoriesManagementComponent implements OnInit, OnDestroy {
       });
   }
 
+  public handleDeleteCategory(category: TransactionCategory): void {
+    this.pagesTransactionCategoriesService.getTransactionCategoryById(category.id!).subscribe((data) => {
+      const isAssigned = this.isCategoryAssignedToTransactions(data);
+
+      if(isAssigned) {
+        this.handleCategoryDeletingWhenNotAssigned(data);
+      } else {
+        this.handleCategoryDeletingWhenAssigned(category);
+      }
+    });    
+  }
+
+  public handleCategoryDeletingWhenAssigned(category: TransactionCategory): void {
```

Comment:
I think it will be much cleaner if we will always show just one button. You can pass to deletion confirmation modal flag like `categoryAssignedToTransactions` and based on that show dropdown and text or not.